src/serverenvironment.cpp

```diff
@@ -1010,21 +1010,21 @@ u8 ServerEnvironment::findSunlight(v3s16 pos)
 	u8 found_light = 0;
 	struct stack_entry {
 		v3s16 pos;
-		s8 dist;
+		s16 dist;
 	};
-	// TODO: dynamic stack size
-	struct stack_entry *stack = new struct stack_entry[100];
-	stack[0] = {pos, 0};
-	int sp = 1;
-	// TODO: find that hash map which works good for positions near each other
-	// (and bad for positions spread everywhere around the map)
+	std::stack<stack_entry> stack;
+	stack.push({pos, 0});
```

Comment:
why a stack and not a queue ?

Reply:
It uses depth first search to find the sun light, so a queue isn't needed. By default [stack uses the dequeue as container](http://en.cppreference.com/w/cpp/container/stack).